src/pystow/utils.py

```diff
@@ -48,8 +122,15 @@ def download(
     :raises Exception: Thrown if an error besides a keyboard interrupt is thrown during download
     :raises KeyboardInterrupt: If a keyboard interrupt is thrown during download
     :raises ValueError: If an invalid backend is chosen
+    :raises HexDigestError: If the downloaded file's hexdigest does not match
     """
+    # input normalization
+    path = Path(path).resolve()
+
+    skip_download = False
     if os.path.exists(path) and not force:
+        skip_download = not get_offending_hexdigests(destination=path, hexdigests=hexdigests)
```

Comment:
I think we need to discuss what should happen if it's got a mismatch. Should it re-download, or should it throw an exception?

Reply:
I guess we can just fail fast. The user can provide `force=True` to get a fresh version of the file.